redfish-core/lib/account_service.hpp

```diff
@@ -1377,6 +1377,141 @@ inline void updateUserProperties(std::shared_ptr<bmcweb::AsyncResp> asyncResp,
+        },
+        "xyz.openbmc_project.Certs.ACF."
+        "Manager",
+        "/xyz/openbmc_project/certs/ACF", "xyz.openbmc_project.Certs.ACF",
+        "InstallACF", decodedAcf);
+}
+
+inline void triggerUnauthenticatedACFUpload(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    std::optional<nlohmann::json> oem)
+{
+    std::optional<nlohmann::json> ibm;
+    if (!redfish::json_util::readJson(*oem, asyncResp->res, "IBM", ibm))
+    {
+        BMCWEB_LOG_ERROR << "Illegal Property ";
+        messages::propertyMissing(asyncResp->res, "IBM");
+        return;
+    }
+
+    if (ibm)
+    {
+        std::optional<nlohmann::json> acf;
+        if (!redfish::json_util::readJson(*ibm, asyncResp->res, "ACF", acf))
+        {
+            BMCWEB_LOG_ERROR << "Illegal Property ";
+            messages::propertyMissing(asyncResp->res, "ACF");
+            return;
+        }
+
+        if (acf)
+        {
+            std::vector<uint8_t> decodedAcf;
+            std::optional<std::string> acfFile;
+            if (acf.value().contains("ACFFile") &&
+                acf.value()["ACFFile"] == nullptr)
+            {
+                acfFile = "";
+            }
+            else
+            {
+                if (!redfish::json_util::readJson(*acf, asyncResp->res,
+                                                  "ACFFile", acfFile))
+                {
+                    BMCWEB_LOG_ERROR << "Illegal Property ";
+                    messages::propertyMissing(asyncResp->res, "ACFFile");
+                    return;
+                }
+
+                std::string sDecodedAcf;
+                if (!crow::utility::base64Decode(*acfFile, sDecodedAcf))
+                {
+                    BMCWEB_LOG_ERROR << "base64 decode failure ";
+                    messages::internalError(asyncResp->res);
+                    return;
+                }
+                try
+                {
+                    std::copy(sDecodedAcf.begin(), sDecodedAcf.end(),
+                              std::back_inserter(decodedAcf));
+                }
+                catch (const std::exception& e)
+                {
+                    BMCWEB_LOG_ERROR << e.what();
+                    messages::internalError(asyncResp->res);
+                    return;
+                }
+            }
+
+            crow::connections::systemBus->async_method_call(
+                [asyncResp, decodedAcf](const boost::system::error_code ec,
+                                        const std::variant<bool>& retVal) {
+                    if (ec)
+                    {
+                        BMCWEB_LOG_ERROR
+                            << "Failed to read ACFWindowActive property";
+                        messages::internalError(asyncResp->res);
+                        return;
+                    }
+
+                    const bool* isACFWindowActive = std::get_if<bool>(&retVal);
+
+                    if (isACFWindowActive == nullptr)
```

Comment:
Question: Is com.ibm.PanelApp  guaranteed to be running?  Or is it running only when an OpPanel is installed?
Would it be better to treat this condition "(isACFWindowActive == nullptr)" as if "*isACFWindowActive == false"?

Reply:
No, com.ibm.PanelApp is only available when panel app is running. Hence it is not guaranteed that we will get the value always.

Reply:
If I understand correctly, in the following scenario:
- The LCD screen in not installed, so the Panel App is not running.
- Someone attempts an unauthenticcated ACF upload, and gets to this function with isACFWindowActive == nullptr
- That user gets internalError.  <-- That HTTP status seems wrong.

I think in this scenario we must handle (isACFWindowActive == nullptr) the same as (*isACFWindowActive == false).  This results in the user getting insufficientPrivilege

Reply:
I am sorry, I mentioned wrongly before, Panel app will be running irrespective of panel being connected to the system.

Reply:
In that case, we should always get the value of isACFWindowActive, and will only get isACFWindowActive=nullptr if the BMC is not working correctly.  The code looks good as it is.